nifi-commons/nifi-write-ahead-log/src/main/java/org/wali/MinimalLockingWriteAheadLog.java

```diff
@@ -360,8 +369,13 @@ private Long recoverFromSnapshot(final Map<Object, T> recordMap) throws IOExcept
 
         // at this point, we know the snapshotPath exists because if it didn't, then we either returned null
         // or we renamed partialPath to snapshotPath. So just Recover from snapshotPath.
-        try (final DataInputStream dataIn = new DataInputStream(new BufferedInputStream(Files.newInputStream(snapshotPath, StandardOpenOption.READ)))) {
-            final String waliImplementationClass = dataIn.readUTF();
+        try (final DataInputStream dataIn = new DataInputStream(new BufferedInputStream(SimpleCipherInputStream.wrapWithKey(Files.newInputStream(snapshotPath, StandardOpenOption.READ), cipherKey)))) {
+            String waliImplementationClass;
+            try {
+                waliImplementationClass = dataIn.readUTF();
+            } catch (final java.io.UTFDataFormatException e) {
+                throw new IOException("malformed input");
```

Comment:
I don't have a problem with wrapping the Exception with an IOException, but the error message provided should provide more detail about what is happening, such as `"Unable to read Write-Ahead Log implementation class from " + snapshotPath` and should include the `UTFDataFormatException` as the cause of the Exception

Reply:
This is really a great catch, thank you for seeing this.  I've updated the message and included `snapshotPath` and the original exception.